src/casp/services/api/routers/cell_operations_router.py

```diff
@@ -31,8 +30,44 @@ async def annotate(
 
     :return: JSON response with annotations.
     """
+
+    return await cell_operations_service.annotate_adata_file(
+        user=request_user,
+        file=file.file,
+        model_name=model_name,
+        include_dev_metadata=include_dev_metadata,
+        consensus_strategy=ConsensusStrategyType.CELL_TYPE_STATISTICS,
+    )
+
+
+@cell_operations_router.post(
+    "/annotate-ontology-aware-strategy",
+    response_model=schemas.QueryAnnotationOntologyAwareType,
+)
+async def annotate_ontology_aware_strategy(
+    file: UploadFile = File(),
+    model_name: str = Form(),
+    normalize: t.Optional[bool] = Form(default=None),
+    request_user: models.User = Depends(dependencies.authenticate_user),
+):
+    """
+    Annotate a single anndata file with Cellarium CAS. Input file should be validated and sanitized according to the
+    model schema.
+
+    :param file: Byte object of :class:`anndata.AnnData` file to annotate.
+    :param model_name: Model name to use for annotation. See `/list-models` endpoint for available models.
+    :param normalize: Boolean flag indicating whether to normalize the data before annotation.
+    :param request_user: Authorized user object obtained  by token from `Bearer` header.
+
+    :return: JSON response with annotations.
+    """
+
     return await cell_operations_service.annotate_adata_file(
-        user=request_user, file=file.file, model_name=model_name, include_dev_metadata=include_dev_metadata
+        user=request_user,
+        file=file.file,
+        model_name=model_name,
+        consensus_strategy=ConsensusStrategyType.ONTOLOGY_AWARE,
+        normalize=normalize,
```

Comment:
I suggest dropping `normalize` altogether for clarity. Also, the docstring is wrong (`normalize the data before annotation`), which raised my eyebrows for a second :-)

We want the scores, by definition, to range in [0, 1]. It might be a good idea to include `total_weight` as a part of the output in the dictionary:
```
{
    ontology_term_scores = {
        CL_... = ...,
        CL_... = ...,
        CL_... = ...,
        CL_... = ...,
    },
    misc = {
        total_weight = ...,
    }
}
```